### PR TITLE
Hook up BlockRateController to Pacemaker

### PR DESCRIPTION
**Pull request by jordanschalm**
This PR hooks up the BlockRateController to the Pacemaker. 
- `BlockRateDelay` config field is replaced by `ProposalDurationProvider`:
  - For collection nodes, a static provider is used (little code changes and same behaviour as before)
  - For consensus nodes, the BlockRateController is passed as the provider
- Adds flags for all cruisectl parameters

### Comments
- There are several rough edges in this PR -- I've kept changes minimal in expectation of conflicts with Alex's proposal for changes
- I think we would like to remove/rename `--block-rate-delay`. However, many environments are configured to use this flag, including all our networks, integration tests, and (I believe) partners. In particular for partners, I'm hoping to make a backward compatible change:
  - If you specify `--block-rate-delay` and no `cruise-ctl-*` flags, we will translate the flag value into `DefaultProposalDuration`
  - If you `--block-rate-delay` and any `cruise-ctl` flags, we will throw an error
- Remaining TODO: add admin commands for new flags

### CODE REVIEW

**Review by AlexHentschel — APPROVED** [review]
👍

**Review by AlexHentschel — APPROVED** [review]
👍

File: consensus/hotstuff/pacemaker/pacemaker.go [hunk 1]
```diff
@@ -86,7 +89,10 @@ func (p *ActivePaceMaker) LastViewTC() *flow.TimeoutCertificate { return p.viewT
 func (p *ActivePaceMaker) TimeoutChannel() <-chan time.Time { return p.timeoutControl.Channel() }
 
 // BlockRateDelay returns the delay for broadcasting its own proposals.
-func (p *ActivePaceMaker) BlockRateDelay() time.Duration { return p.timeoutControl.BlockRateDelay() }
+// todo rename?
+func (p *ActivePaceMaker) BlockRateDelay() time.Duration {
```
Comment (durkmurder) [inline consensus/hotstuff/pacemaker/pacemaker.go:93]: At this point I would lean into removing the `BlockRateDelay` completely from pacemaker and using provider in `EventHandler`. Pacemaker has nothing to do with it and just proxies call to provider, it has zero overlap with logic of pacemaker, previously we at least shared a config but not anymore.

File: consensus/hotstuff/pacemaker/pacemaker_test.go [hunk 2]
```diff
@@ -403,29 +399,48 @@ func (s *ActivePaceMakerTestSuite) Test_Initialization() {
...
 }
 
+type dynamicProposalDurationProvider struct {
+	dur time.Duration
+}
+
+func (p *dynamicProposalDurationProvider) ProposalDuration() time.Duration {
+	return p.dur
+}
+
```
Comment (durkmurder) [inline consensus/hotstuff/pacemaker/pacemaker_test.go:432]: lets make it obvious that we implement a specific interface

**Review by durkmurder — APPROVED** [review]
Looks good, suddenly I want to decouple block rate delay from pacemaker, which will simplify code a bit as well. We can do it later as well, let me know your thoughts on this.

**Comment by jordanschalm** [comment]
> suddenly I want to decouple block rate delay from pacemaker, which will simplify code a bit as well. We can do it later as well

I was thinking the same actually while creating this PR. I didn't mainly to avoid conflicts with Alex's revisions. After reviewing his changes, I'm still also leaning toward separating them. Not high on my list of priorities to get in this week, but I'm onboard with the suggestion 👍

File: consensus/hotstuff/pacemaker/pacemaker_test.go (hunk 2 above)
Comment (jordanschalm) [reply consensus/hotstuff/pacemaker/pacemaker_test.go:432]: I think alex has done this in his revision

**Comment by jordanschalm** [comment]
bors merge

**Pull request merged**